### PR TITLE
Added work of native back button

### PR DESCRIPTION
**Pull request by IraSoro**

### CODE REVIEW

**Review by imblowfish — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -53,7 +53,7 @@ import { CyclesContext } from './state/Context';
 
 setupIonicReact();
 
-const App: React.FC = () => {
+const PeriApp: React.FC = () => {
```
[the remark this answers is not in the document]
Comment (IraSoro) [reply src/App.tsx:56]: Fixed

File: src/App.tsx [hunk 2]
```diff
@@ -134,5 +134,5 @@ const App: React.FC = () => {
   )
 };
 
-export default App;
+export default PeriApp;
```
[the remark this answers is not in the document]
Comment (IraSoro) [reply src/App.tsx:137]: Fixed too

File: src/modals/CalendarModal.tsx [hunk 3]
```diff
@@ -98,9 +99,7 @@ const CalendarModal = (props: PropsCalendarModal) => {
                     <IonButtons slot="buttons">
                         <IonButton
                             color="basic"
-                            onClick={() => {
-                                props.setIsOpen(false);
-                            }}
+                            onClick={() => props.setIsOpen(false)}
```
[the remark this answers is not in the document]
Comment (IraSoro) [reply src/modals/CalendarModal.tsx:102]: Fixed

File: src/pages/TabHome.tsx [hunk 4]
```diff
@@ -197,11 +208,7 @@ const TabHome = () => {
                 </IonItem>
               </IonCardContent>
             </IonCard>
-            <IonButton onClick={() => setIsInfoModal(true)} class="info-button">learn more about the current state</IonButton>
-            <InfoModal
-              isOpen={isInfoModal}
-              setIsOpen={setIsInfoModal}
-            />
+            <InfoModal isOpen={isInfoModal} setIsOpen={setIsInfoModal} />
```
[the remark this answers is not in the document]
Comment (IraSoro) [reply src/pages/TabHome.tsx:211]: Ok

File: src/modals/InfoModal.tsx [hunk 5]
```diff
@@ -65,38 +65,50 @@ const InfoModal = (props: PropsInfoModal) => {
...
-                        <IonCardHeader class="info">
-                            {phase.title}
-                        </IonCardHeader>
-                        <IonCardContent style={{ textAlign: "justify" }}>
-                            {phase.description}
-                        </IonCardContent>
-                    </IonCard>
-                </div>
+        <>
+            <IonButton
```
[the remark this answers is not in the document]
Comment (IraSoro) [reply src/modals/InfoModal.tsx:69]: Ok. I'll make a formatting task.

File: src/pages/TabHome.tsx [hunk 6]
```diff
@@ -97,49 +100,52 @@ function useDaysBeforePeriod(): DaysBeforePeriod {
...
       .catch((err) => {
         console.error(`Can't get cycles ${(err as Error).message}`);
         setIsWelcomeModal(true);
       });
-
   }, []);
 
+  useEffect(() => {
+    const backButtonHandler = () => {
+      if (isCalendarModal) {
```
[the remark this answers is not in the document]
Comment (IraSoro) [reply src/pages/TabHome.tsx:126]: Code fixed, good idea. But the transition to the "/home" is needed, without it it does not work. This is a feature of the native back button.

File: src/pages/TabHome.tsx [hunk 7]
```diff
@@ -28,6 +27,10 @@ import {
 
 import { get } from '../data/Storage';
 
+import { useHistory } from 'react-router';
```
[the remark this answers is not in the document]
Comment (IraSoro) [reply src/pages/TabHome.tsx]: Fixed

File: src/pages/TabHome.tsx [hunk 8]
```diff
@@ -97,49 +100,52 @@ function useDaysBeforePeriod(): DaysBeforePeriod {
...
         setIsWelcomeModal(true);
       });
-
   }, []);
 
+  useEffect(() => {
+    const backButtonHandler = () => {
+      if (isCalendarModal) {
+        setIsCalendarModal(false);
+        history.push('/home');
```
[the remark this answers is not in the document]
Comment (IraSoro) [reply src/pages/TabHome.tsx]: Changed to `useIonRouter`

**Review by imblowfish — APPROVED** [review]

**Comment by imblowfish** [comment]
I fixed merge conflicts, @IraSoro could you please verify that all ok now?

**Pull request merged**